### PR TITLE
Added ability to configure unique keys for collection

### PR DESCRIPTION
**Pull request by florisrobbemont**
We needed to add a unique key policy when creating the collection. Since we like to let Cosmonaut handle collection provisioning I added this feature to the codebase.

### CODE REVIEW

**Comment by Elfocrash** [comment]
Hello. The code looks good and I like the idea of the feature but before I approve this I need to test how the shared collection behaviour would work with this.

**Comment by florisrobbemont** [comment]
I’ve built that into the code and into the tests. You cannot have a unique key policy for a shared collection, since not all of the entities might have those unique properties.
I also can’t think of a scenario where this would be useful in a shared collection.

So id you configure an entity with a unique key and a shared collection attribute it will throw an exception.

Would like to hear your thoughts on this.

**Comment by Elfocrash** [comment]
What would the behaviour be if the collection exists with a different unique key policy?

**Comment by florisrobbemont** [comment]
In that case it would also crash on initialization:

https://github.com/florisrobbemont/Cosmonaut/blob/0c1b48cea9b2c016f551e019f8dde7a8b514d47a/src/Cosmonaut/Storage/CosmosCollectionCreator.cs#L49

The check is in the CollectionCreator, before it goes into the CosmosClient.
So if you try to configure this on an existing collection/entitity you will get an exception on init.

**Comment by Elfocrash** [comment]
I would rather have it behave differently. If the collection doesn't exist then create it with the uniquekeypolicy. If the collection does exist then query then don't do anything. It should go on par with the indexing policy and the RUs.

**Comment by florisrobbemont** [comment]
I've ran some tests, and Cosmos DB accepts it when you push a unique key that doesn't exists. I don't know how this will hold up in runtime, since the non existing property will probably result in 'null', and will therefor not be unique accross the documents.

The code doesn't throw an exception anymore for both existing and non-existing collection, leaving it up to the user to make the correct decisions.

File: src/Cosmonaut/CosmosStoreSettings.cs [hunk 1]
```diff
@@ -74,7 +78,8 @@ public class CosmosStoreSettings
             string authKey,
             ConnectionPolicy connectionPolicy = null,
             IndexingPolicy indexingPolicy = null,
-            int defaultCollectionThroughput = CosmosConstants.MinimumCosmosThroughput)
+            int defaultCollectionThroughput = CosmosConstants.MinimumCosmosThroughput,
+            UniqueKeyPolicy uniqueKeyPolicy = null)
```
Comment (Elfocrash) [inline src/Cosmonaut/CosmosStoreSettings.cs]: Can you remove that from the ctor? I am moving away from optional parameters via the ctor

File: src/Cosmonaut/CosmosStoreSettings.cs [hunk 2]
```diff
@@ -58,13 +60,15 @@ public class CosmosStoreSettings
...
```
Comment (Elfocrash) [inline src/Cosmonaut/CosmosStoreSettings.cs]: Can you remove that from the ctor? I am moving away from optional parameters via the ctor

Comment (florisrobbemont) [reply src/Cosmonaut/CosmosStoreSettings.cs]: No problem. I just followed the patterns that were already there.
I’ll remove it tomorrow.

File: tests/Cosmonaut.System/CosmosProvisioningTests.cs [hunk 3]
```diff
@@ -153,6 +156,50 @@ public async Task CosmosStoreSettings_CreatesCollectionWithThroughput_WhenThroug
             collectionOffer.Content.OfferThroughput.Should().Be(10000);
         }
 
+        [Fact]
+        public async Task CosmosStoreSettings_CreatesCollectionWithUniqueKeyPolicy()
+        {
+            var cosmosStoreSettings = new CosmosStoreSettings(_databaseId, _emulatorUri, _emulatorKey, settings =>
```
Comment (Elfocrash) [inline tests/Cosmonaut.System/CosmosProvisioningTests.cs]: Indentation looks weird on this test. Can you fix the formatting please? I'm about to merge this.

**Pull request merged**